router/core/src/main/scala/io/buoyant/router/DiscardingFactoryToService.scala

```diff
@@ -0,0 +1,99 @@
+package io.buoyant.router
+
+import com.twitter.finagle._
+import com.twitter.util.{Future, Time}
+import io.buoyant.router.DiscardingFactoryToService.RequestDiscarder
+
+/**
+ * Turns a [[com.twitter.finagle.ServiceFactory]] into a
```

Comment:
We should add an attribution comment saying this code was copied+modified from finagle's FactoryToService.